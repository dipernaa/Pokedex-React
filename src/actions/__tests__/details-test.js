import mockStore from 'redux-mock-store';
import detailActions from '../details';

jest.mock('../utils/asyncUtils', () => ({
  buildActionNames: (baseName) => ({
    failure: `${baseName}_FAILURE`,
    request: `${baseName}_REQUEST`,
    success: `${baseName}_SUCCESS`
  }),
  createAsyncAction: (actionNames, asyncFn) => (
    asyncFn(mockReduxStore.getState())
      .then(result => ({
        actionNames,
        result
      }))
  )
}));

jest.mock('../utils/api', () => ({
  getAsync: (endpoint) => Promise.resolve({
    endpoint
  })
}));

const mockReduxStore = mockStore();

beforeEach(() => {
  mockReduxStore.clearActions();
});

it('creates a get pokemon details action', () =>
  detailActions.getPokemonDetails('http://www.testurl.com')
    .then(result => {
      expect(result).toMatchSnapshot();
    })
);
