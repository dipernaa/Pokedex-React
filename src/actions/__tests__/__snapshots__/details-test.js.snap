// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a get pokemon details action 1`] = `
Object {
  "actionNames": Object {
    "failure": "GET_POKEMON_DETAILS_FAILURE",
    "request": "GET_POKEMON_DETAILS_REQUEST",
    "success": "GET_POKEMON_DETAILS_SUCCESS",
  },
  "result": Object {
    "endpoint": "/api/details?url=http://www.testurl.com",
  },
}
`;
